test/hotspot/gtest/riscv/test_assembler_riscv.cpp

```diff
@@ -260,4 +260,133 @@ TEST_VM(RiscV, cmpxchg_int32_plain_maybe_zacas) {
   }
 }
 
+template <typename TESTSIZE, Assembler::operand_size ASMSIZE>
+class WeakCmpxchgTester {
+ public:
+  typedef TESTSIZE (*cmpxchg_narrow)(intptr_t addr, TESTSIZE expected, TESTSIZE new_value, TESTSIZE result,
+                                   int64_t scratch0, int64_t scratch1, int64_t scratch2);
+  
+  typedef TESTSIZE (*cmpxchg_func)(intptr_t addr, TESTSIZE expected, TESTSIZE new_value, TESTSIZE result);
+
+  static TESTSIZE weak_narrow_cmpxchg(intptr_t addr, TESTSIZE expected, TESTSIZE new_value) {
+    BufferBlob* bb = BufferBlob::create("riscvTest", 128);
+    CodeBuffer code(bb);
+    MacroAssembler _masm(&code);
+    address entry = _masm.pc();
+    {
+       _masm.weak_cmpxchg_narrow_value(/*addr*/ c_rarg0, /*expected*/ c_rarg1, /*new_value*/ c_rarg2,
+                                      ASMSIZE, Assembler::relaxed, Assembler::relaxed,
+                                      /*result*/ c_rarg3, c_rarg4, c_rarg5, c_rarg6); /* Uses also t0-t1, caller saved */
+      _masm.mv(c_rarg0, c_rarg3);
+      _masm.ret();
+    }
+    _masm.flush(); // icache invalidate
+    TESTSIZE ret = ((cmpxchg_narrow)entry)(addr, expected, new_value, /*result*/ 67, -1, -1, -1);
+    BufferBlob::free(bb);
+    return ret;
+  }
+  
+  static TESTSIZE weak_cmpxchg(intptr_t addr, TESTSIZE expected, TESTSIZE new_value) {
+    BufferBlob* bb = BufferBlob::create("riscvTest", 128);
+    CodeBuffer code(bb);
+    MacroAssembler _masm(&code);
+    address entry = _masm.pc();
+    {
+       _masm.cmpxchg_weak(/*addr*/ c_rarg0, /*expected*/ c_rarg1, /*new_value*/ c_rarg2,
```

Comment:
Seems that we should rename macro-assembler routine `cmpxchg_weak` to `weak_cmpxchg`.
I have prepared another PR to fix this naming issue: https://github.com/openjdk/jdk/pull/22505.

Reply:
Fixed

Reply:
Ok!